app/forms/routing/rate_group_duplicator_form.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class Routing::RateGroupDuplicatorForm < ApplicationForm
+  attribute :name, :string
+  attribute :id, :integer
+
+  validates :name, :id, presence: true
+
+  validate do
+    errors.add(:id, :invalid) unless Routing::RateGroup.exists?(id)
+    errors.add(:name, :taken) if Routing::RateGroup.exists?(name: name)
+  end
+
+  private
+
+  def _save
+    dst = Routing::RateGroup.create!(
+      name: name
+    )
+    src = Routing::RateGroup.find(id)
```

Comment:
create method `src_rate_group` that caches instance of `Routing::RateGroup` similar to `JsonApiModel::OriginatedCps#account`

use it in save method and in validator on line 10